src/core/hle/D3D8/Direct3D9/Direct3D9.cpp

```diff
@@ -1685,6 +1686,53 @@ uint8_t *ConvertD3DTextureToARGB(
 
 extern void HLE_init_pgraph_plugins(); // implemented in XbPushBuffer.cpp
 
+// FIXME move cache functions elsewhere
+std::filesystem::path GetVshCachePath() {
+	namespace fs = std::filesystem;
+
+	// Get unique title string
+	// Hash the loaded XBE's header, use it as a filename
+	uint64_t uiHash = ComputeHash((void*)&CxbxKrnl_Xbe->m_Header, sizeof(Xbe::Header));
+	std::stringstream sstream;
+	std::wstring wTitleName = CxbxKrnl_Xbe->m_Certificate.wsTitleName;
+	std::string titleName = std::string(wTitleName.begin(), wTitleName.end());
+	CxbxKrnl_Xbe->PurgeBadChar(titleName);
+
+	std::stringstream fileName;
+	fileName << titleName << "-" << std::hex << uiHash << ".vscache";
+
+	return fs::path(g_DataFilePath) / "Shaders" / fileName.str();
```

Comment:
just a nit change here:
```suggestion
	return fs::path(g_DataFilePath) / "ShaderCache" / fileName.str();
```